lib/serverless/api-gateway.js

```diff
@@ -30,11 +32,53 @@ class LambdaProxyWebRequest {
  */
 class LambdaProxyWebResponse {
   constructor(lambdaResponse) {
-    this.headers = lambdaResponse.headers
+    this.headers = normalizeHeaders(lambdaResponse)
     this.statusCode = lambdaResponse.statusCode
   }
 }
 
+/**
+ * normalizes query string parameters either from multi value query string parameters or normal query string parameters to a
+ * key map with comma separated strings
+ *
+ * @param {object} event The event with headers to normalize
+ * @returns {Object<string, string>} The normalized headers map
+ */
+function normalizeQueryStringParameters(event) {
+  if (!event.multiValueQueryStringParameters) {
+    return event.queryStringParameters
+  }
+  return Object.fromEntries(
+    Object.entries(event.multiValueQueryStringParameters).map(([param, value]) => {
+      if (Array.isArray(value)) {
+        return [param, value.join(',')]
+      }
+      return [param, value]
+    })
+  )
+}
+
+/**
+ * normalizes headers either from multi value headers or normal headers to a
+ * lowercase key map with comma separated string
+ *
+ * @param {object} event The event with headers to normalize
+ * @param {boolean} lowerCaseKey Whether to lowercase the header names or not
+ * @returns {Object<string, string>} The normalized headers map
+ */
+function normalizeHeaders(event, lowerCaseKey = false) {
+  return Object.fromEntries(
+    Object.entries(event.multiValueHeaders ?? event.headers).map(([headerKey, headerValue]) => {
```

Comment:
From reading the AWS docs/testing, it appears that both `multiValueHeaders` and `headers` can be defined at the same time, so I think we'll need to update this function to try and normalize both keys.

Reply:
for api gateway yes but for alb not, also if you have multiValueHeaders, all the values are the same as in the normal headers key, so no value in it to normalize it also when the multi value headers are send. the output will be even less. 

also what should happen for a header with multiple values and the same header with only a single value

Reply:
Also in the context of a lambda which is moved through the application load balancer, you only have mutliValueHeaders or normal headers but never both.